dis_snek/models/discord/message.py

```diff
@@ -279,10 +282,20 @@ def _process_dict(cls, data: dict, client: "Snake") -> dict:
                 mention_ids.append(client.cache.place_user_data(user_data).id)
         data["mention_ids"] = mention_ids
 
-        if "mention_channels" in data:
+        extra = channel_mention.findall(data["content"])
+        if "mention_channels" in data or len(extra) > 0:
             mention_channels = []
             for channel_data in data["mention_channels"]:
                 mention_channels.append(ChannelMention.from_dict(channel_data, client))
+            for channel_id in extra:
+                if channel := client.get_channel(channel_id):
+                    channel_data = {
+                        "id": channel.id,
+                        "guild_id": channel._guild_id,
+                        "type": channel.type,
+                        "name": channel.name,
+                    }
+                    mention_channels.append(ChannelMention.from_dict(channel_data, client))
```

Comment:
hmm will it cause possible duplicated `mention_channels` entries? If it is already in `data["mention_channels"]` and you parsed it again with regex.